examples/jsm/loaders/obj2/worker/parallel/OBJLoader2Parser.js

```diff
@@ -130,60 +165,129 @@ OBJLoader2Parser.prototype = {
 
 	},
 
+	/**
+	 * Register a function that is called once an asset (mesh/material) becomes available.
+	 *
+	 * @param onAssetAvailable
+	 * @return {OBJLoader2Parser}
+	 */
 	setCallbackOnAssetAvailable: function ( onAssetAvailable ) {
 
-		if ( onAssetAvailable !== null && onAssetAvailable !== undefined ) {
+		if ( onAssetAvailable !== null && onAssetAvailable !== undefined && onAssetAvailable instanceof Function ) {
 
 			this.callbacks.onAssetAvailable = onAssetAvailable;
 
 		}
+		return this;
 
 	},
 
+	/**
+	 * Register a function that is used to report overall processing progress.
+	 *
+	 * @param {Function} onProgress
+	 * @return {OBJLoader2Parser}
+	 */
 	setCallbackOnProgress: function ( onProgress ) {
 
-		if ( onProgress !== null && onProgress !== undefined ) {
+		if ( onProgress !== null && onProgress !== undefined && onProgress instanceof Function ) {
 
 			this.callbacks.onProgress = onProgress;
 
 		}
+		return this;
 
 	},
 
+	/**
+	 * Register an error handler function that is called if errors occur. It can decide to just log or to throw an exception.
+	 *
+	 * @param {Function} onError
+	 * @return {OBJLoader2Parser}
+	 */
 	setCallbackOnError: function ( onError ) {
 
-		if ( onError !== null && onError !== undefined ) {
+		if ( onError !== null && onError !== undefined && onError instanceof Function ) {
 
 			this.callbacks.onError = onError;
 
 		}
+		return this;
 
 	},
 
-	setLogging: function ( enabled, debug ) {
+	/**
+	 * Announce feedback which is give to the registered callbacks.
+	 * @private
+	 *
+	 * @param {string} type The type of event
+	 * @param {string} text Textual description of the event
+	 * @param {number} numericalValue Numerical value describing the progress
+	 */
+	_onProgress: function ( type, text, numericalValue ) {
+
+		let message = text ? text : '';
+		let event = {
```

Comment:
It seems `event` is never used in `_onProgress()`. If it is not required, the object should be deleted.

Reply:
Yes, you are right. The signature of the function should then be simplified along. Shall I submit a new PR?

Reply:
Done, see below.